### PR TITLE
Docs/deep dive improvements

### PR DESCRIPTION
**Pull request by bermi**
Fixing issue found while going through the deep dive docs.

I went through the deep dive on an intel macOS, and it worked fine. I didn't add macOS-specific instructions since I do not have an arm machine, and I wasn't sure if it would work there.

### CODE REVIEW

File: scripts/upgrade_ubuntu_kernel.sh [hunk 1]
```diff
@@ -8,6 +8,15 @@ if [ "$DISTRIBUTION" != "Ubuntu" ]; then
     exit 1
 fi
 
+# No need to downgrade the kernel if we are on a newer version.
+CURRENT_KERNEL_VERSION=$(uname -r | grep -o "[0-9\.]*" | head -1)
+if [ "$(printf '%s\n' "5.7.15" "$CURRENT_KERNEL_VERSION" | sort -V | head -n1)" = "5.7.15" ]; then
+    echo "Current kernel ${CURRENT_KERNEL_VERSION} supports io_uring. Not need to upgrade."
```
Comment (jorangreef) [inline scripts/upgrade_ubuntu_kernel.sh]:
```suggestion
    echo "Current kernel ${CURRENT_KERNEL_VERSION} supports io_uring. No need to upgrade."
```

File: docs/DEEP_DIVE.md [hunk 2]
```diff
@@ -26,11 +32,11 @@ scripts/install.sh
 
 ## Simulate non-pristine lab conditions
 
-You are ready to rock! To simulate non-pristine lab conditions (tiger beetles thrive in harsh environments with noisy neighbors) turn up your favorite album, or if you're looking for something new we suggest Noel Gallagher's High Flying Birds.
+You are ready to rock! To simulate non-pristine lab conditions (tiger beetles thrive in harsh environments with noisy neighbors), turn up your favorite album, or if you're looking for something new, we suggest Noel Gallagher's High Flying Birds.
 
 Be aware that super non-pristine lab conditions such as an active video call might cause cache pollution and drop throughput. But rock 'n roll is generally fine.
 
-## Benchmark!
+## Benchmark
```
Comment (jorangreef) [inline docs/DEEP_DIVE.md]:
```suggestion
## Benchmark!
```

File: docs/DEEP_DIVE.md [hunk 3]
```diff
@@ -42,15 +48,17 @@ scripts/benchmark.sh
 
 Take a look at the source code of these demos before you run them. Check out our screencast of these demos for much more detail (note that there have been a few changes to our types since the screencast).
 
-Let's turn up the log level some more (and your favorite album) so you can see everything the server does as you run these demos:
+Let's turn up the log level some more (and your favorite album), so you can see everything the server does as you run these demos:
+
+- Open `src/config.zig` in your editor and change `log_level` to `3` (debug).
 
-* Open `src/config.zig` in your editor and change `log_level` to `3` (debug).
+- Rebuild tigerbeetle using the new debug log level by running `zig/zig build -Dcpu=baseline -Drelease-safe && mv zig-out/bin/tigerbeetle .`
```
Comment (jorangreef) [inline docs/DEEP_DIVE.md]:
```suggestion
- Rebuild TigerBeetle using the new debug log level by running `zig/zig build -Dcpu=baseline -Drelease-safe && mv zig-out/bin/tigerbeetle .`
```

File: docs/DEEP_DIVE.md [hunk 4]
```diff
@@ -42,15 +48,17 @@ scripts/benchmark.sh
...
 
-* Start a single replica cluster:
+- Start a single replica cluster:
 Init:
 `./tigerbeetle init --cluster=1 --replica=0 --directory=.`
 Run:
-`./tigerbeetle start --cluster=1 --replica=0 --addresses=3001 --directory=. &`
+`./tigerbeetle start --cluster=1 --replica=0 --addresses=3001 --directory=. 2>&1 | grep -v vsr &`
```
Comment (jorangreef) [inline docs/DEEP_DIVE.md]:
```suggestion
`./tigerbeetle start --cluster=1 --replica=0 --addresses=3001 --directory=. &`
```

**Comment by jorangreef** [comment]
On fire!

File: docs/DEEP_DIVE.md (hunk 2 above)
Comment (bermi) [reply docs/DEEP_DIVE.md]: Sorry about that one, I should disable auto-lint for markdown on my editor

File: docs/DEEP_DIVE.md (hunk 4 above)
Comment (bermi) [reply docs/DEEP_DIVE.md]: The log at the documented level becomes impractical with the flood of vsr messages

**Comment by bermi** [comment]
BTW, let me squash these changes into a single commit before merging it

**Pull request merged**